docs/docs/adding-app-and-website-functionality.md

```diff
@@ -0,0 +1,78 @@
+
+When you export a React component from a file in the `src/pages` directory (in this case `src/pages/home.js`) Gatsby will automatically create a static page. By looping through markdown files in your filesystem you can create pages for all blog posts programmatically. The docs have more informataion about [creating and modifying pages](/docs/creating-and-modifying-pages/).
+
+These created pages _could_ run JavaScript once React hydrates them, but they don't need to.
+
+### Hybrid app pages
+
+Your created pages can make calls to external services and APIs in order to allow more interactive and dynamic behavior. These are sometimes referred to as hybrid app pages because they share a combination of static features that Gatsby creates to help load your site quickly and performantly as well as traditional web app features.
+
+The following diagram shows a similar site to the one in the previous example, but with a subscription form added to the home page that makes `POST`'s to a database somewhere, as well as blog posts that `GET` recommendations from a database or other API.
+
+![Hybrid Site diagram with pages hitting other services with JavaScript](./images/simple-hybrid-site.png)
+
+Following a pattern like this means you are relying on a backend to remain operational for features like email signups and blog recommendations, but because the static assets created aren't generated by the server on demand, the content on your site (like your blog posts or home page) will never go down and become unavailable.
+
+### Client only routes
+
+Using a React based router is also supported by Gatsby. This pattern is often referred to as client only routes, which are routes not reflected in your statically rendered files.
+
+With Gatsby, you can import a router and set up routes for navigation the same way you would in traditional React apps. The only difference is Gatsby doesn't build those routes into individual pages in the `/public` folder. As a result, in order to allow users to access that URL directly, you can use a plugin to create those pages. This is covered in the [Client Only Routes](/docs/client-only-routes-and-user-authentication/) guide.
+
+The following diagram shows how a `<Router />` component can be mounted on a page. In this example, `src/pages/app` references `<Route />`s.
+
+![Client Only Routes Site diagram with pages setup using a router](./images/simple-client-only-routes.png)
+
+In this illustration, a user page could display specific information about the logged in user, and dynamic routes at `/app/tasks/:id` could display specific information for a task of a given id.
+
+---
+
+Generating performant sites with statically rendered assets is a core focus of Gatsby, but it's only one side of the coin. In this section of the docs, you will find a showcase of guides and concepts on how to level up your site to include all the app-like features on top of the static base.
+
+<GuideList slug={props.slug} />
```

Comment:
This guidelist gets lost at the bottom of the page, and it sorta breaks the pattern of a Reference Guide Overview page. The "In this section" heading doesn't seem to end up in the TOC, either. We have a few options here, let me know if you have any further ideas:

- Turn this page back into an overview page and introduce new content as an "Introduction" or specifically-titled subpage
- Add the guidelist heading to the TOC somehow
- Move the guidelist to the top, above the content? (not ideal as it's super long and likely to get longer)

Reply:
(After me and Marcy discussed this we decided to go with No. 2:  

> Add the guidelist heading to the TOC somehow

so that this PR can keep moving forward, this is the issue: #19291)